x-pack/plugins/upgrade_assistant/public/application/components/tabs/checkup/constants.tsx

```diff
@@ -6,8 +6,8 @@
 
 import { IconColor } from '@elastic/eui';
 import { invert } from 'lodash';
-
-import { DeprecationInfo } from 'src/legacy/core_plugins/elasticsearch';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import type { DeprecationInfo } from '../../../../../../../../src/core/server/elasticsearch/legacy/api_types';
```

Comment:
There was a few references to `DeprecationInfo` in the `upgrade_assistant` plugin. To avoid exporting additional deprecated types from `core`, and knowing that the legacy client is deprecated anyway, I used the internal path instead.